src/bin/pgcopydb/env_utils.c

```diff
@@ -116,6 +116,44 @@ get_env_copy_with_fallback(const char *name, char *result, int maxLength,
 	return true;
 }
 
+/*
+ * get_env_dup copies the environment variable with "name" into
+ * the result buffer using strdup. It returns false when it fails. If the environment
+ * variable is not set the fallback string will be written in the buffer.
+ * Except when fallback is NULL, in that case an error is returned.
+ */
+bool
+get_env_dup(const char *name, char **result, const char *fallback)
```

Comment:
Can we have the same API as the copy based function, meaning `get_env_dup` and `get_env_dup_with_fallback`?

Reply:
Kept just one, because there no `maxLength` that is involved now. So even if its without fallback, one can pass in NULL to `get_env_dup`

Reply:
I still think we should offer both APIs, same as the copy stuff. It's easier to understand, remember, and deal with.